tests/unit/core/server/testServer.js

```diff
@@ -83,9 +110,112 @@ describe('core/server/base', () => {
       );
 
       const response = await testClient({ stubRoutes })
-        .get('/en-US/firefox/simulation-of-a-non-existent-page').end();
+        .get('/en-US/firefox/simulation-of-a-non-existent-page')
+        .end();
 
       expect(response.statusCode).toEqual(404);
     });
+
+    it('does not dispatch setAuthToken() if cookie is not found', async () => {
+      const { store, sagaMiddleware } = createStoreAndSagas({
+        reducers: { reduxAsyncConnect, api: apiReducer },
+      });
+      const response = await testClient({ store, sagaMiddleware })
+        .get('/en-US/firefox/')
+        .end();
+
+      const { api } = store.getState();
+
+      expect(response.statusCode).toEqual(200);
+      expect(api.token).toBe(null);
+    });
+
+    it('dispatches setAuthToken() if cookie is present', async () => {
+      const token = userAuthToken();
+      const { store, sagaMiddleware } = createStoreAndSagas({
+        reducers: { reduxAsyncConnect, api: apiReducer },
+      });
+      const response = await testClient({ store, sagaMiddleware })
+        .get('/en-US/firefox/')
+        .set('cookie', `${defaultConfig.get('cookieName')}="${token}"`)
+        .end();
+
+      const { api } = store.getState();
+
+      expect(response.statusCode).toEqual(200);
+      expect(api.token).toEqual(token);
+    });
+
+    it('runs sagas as early as possible', async () => {
+      const profile = createUserProfileResponse({ id: 42, username: 'babar' });
+
+      mockUserApi
+        .expects('userProfile')
+        .once()
+        .returns(Promise.resolve(profile));
+
+      const token = userAuthToken();
+      const { store, sagaMiddleware } = createStoreAndSagas({
+        reducers: { reduxAsyncConnect, api: apiReducer, user: userReducer },
+      });
+      const response = await testClient({ store, sagaMiddleware, appSagas: userSaga })
+        .get('/en-US/firefox/')
+        .set('cookie', `${defaultConfig.get('cookieName')}="${token}"`)
+        .end();
+
+      const { api, user } = store.getState();
+
+      expect(response.statusCode).toEqual(200);
+      expect(api.token).toEqual(token);
+      expect(user.id).toEqual(42);
+      expect(user.username).toEqual('babar');
+      mockUserApi.verify();
+    });
+
+    it('sends the redux-saga END action when `disableSSR` is `true`', async () => {
```

Comment:
This spec should be more related to the actual feature. I suggest this: "it fetches the user profile even when SSR is disabled"